src/com/android/volley/toolbox/ImageRequest.java

```diff
@@ -141,8 +141,11 @@ private static int getResizedDimension(int maxPrimary, int maxSecondary, int act
                     return Response.error(new ParseError(response));
                 } else {
                     Cache.Entry entry = HttpHeaderParser.parseCacheHeaders(response);
-                    if (entry != null) {
+                    if (entry != null && getTTL() > 0) {
                         entry.setTTL(System.currentTimeMillis() + getTTL());
+                        if (isOfflineCache()) {
+                            entry.keepUntil = entry.ttl;
```

Comment:
we are storing it as an explicit date in the cache, but the to the client it is just a flag. I can modify this behavior, but it didn't really make sense to say keep this for a week, but ttl it for 2 weeks. It doesn't seem like a use case that makes sense.